### PR TITLE
Feature/dont start bittorrent without vpn

### PR DESCRIPTION
**Pull request by grzesiekrzaca**
This is the android portion of #166

### CODE REVIEW

**Comment by gubatron** [comment]
reviewing, will send you a PR with wording changes and a few refactors.

**Comment by gubatron** [comment]
Sent a PR with a couple changes (typo, styling, and wording)

However, I keep getting this crash (even after uninstalling, cleaning project, rebuilding) on Android 6.0

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.frostwire.android/com.frostwire.android.gui.activities.SettingsActivity2}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.support.v7.preference.SwitchPreferenceCompat.setOnPreferenceChangeListener(android.support.v7.preference.Preference$OnPreferenceChangeListener)' on a null object reference
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(ActivityThread.java)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.support.v7.preference.SwitchPreferenceCompat.setOnPreferenceChangeListener(android.support.v7.preference.Preference$OnPreferenceChangeListener)' on a null object reference
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at com.frostwire.android.gui.fragments.preference.ApplicationFragment.setupVPNRequirementOption(ApplicationFragment.java:94)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at com.frostwire.android.gui.fragments.preference.ApplicationFragment.initComponents(ApplicationFragment.java:71)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at com.frostwire.android.gui.views.AbstractPreferenceFragment.onCreatePreferences(AbstractPreferenceFragment.java:54)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.support.v14.preference.PreferenceFragment.onCreate(PreferenceFragment.java:234)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.Fragment.performCreate(Fragment.java:2198)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:942)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1148)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.BackStackRecord.run(BackStackRecord.java:793)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1535)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.FragmentController.execPendingActions(FragmentController.java:325)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.Activity.performStart(Activity.java:6252)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
02-06 11:41:30.731  6509  6509 E AndroidRuntime: 	... 9 more
02-06 11:41:30.732  1591  2200 W ActivityManager:   Force finishing activity com.frostwire.android/.gui.activities.SettingsActivity2
02-06 11:41:30.736  1591  2200 W ActivityManager:   Force finishing activity com.frostwire.android/.gui.activities.MainActivity
```

**Comment by grzesiekrzaca** [comment]
@gubatron fixed the crash you were having.

File: android/res/xml/settings_application.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
             android:summary="@string/bittorrent_network_summary"
             android:title="@string/bittorrent" />
         <android.support.v7.preference.SwitchPreferenceCompat
-            android:key="frostwire.prefs.bittorrent_on_vpn_only"
```
Comment (gubatron) [inline android/res/xml/settings_application.xml]: LOL, I feel stupid :p 

Thank you @grzesiekrzaca

**Comment by gubatron** [comment]
Thank you, my bad.
Testing once more.

**Comment by gubatron** [comment]
Functionality wise this is ready to go.
I've sent a new [PR](https://github.com/grzesiekrzaca/frostwire/pull/5) with experience features.

**Comment by gubatron** [comment]
PR merged. When you have a chance please take a look @aldenml see if there's anything off before merging, this is a power feature.

**Pull request merged**